integrations/shopify/src/setup/register.ts

```diff
@@ -0,0 +1,100 @@
+import { Client } from '@botpress/client'
+import type { IntegrationContext, IntegrationProps } from '@botpress/sdk'
+import axios from 'axios'
+import { ARR_OF_EVENTS, SHOPIFY_API_VERSION } from '../const'
+import type * as botpress from '.botpress'
+import type { Configuration } from '.botpress/implementation/configuration'
+
+type IntegrationLogger = Parameters<IntegrationProps['handler']>[0]['logger']
+type Implementation = ConstructorParameters<typeof botpress.Integration>[0]
+type RegisterFunction = Implementation['register']
+
+function getValue(obj: string | undefined) {
+  if (typeof obj === 'string') {
+    return obj
+  } else {
+    return ''
+  }
+}
+
+export const register: RegisterFunction = async ({
+  client,
+  ctx,
+  logger,
+  webhookUrl,
+}: {
+  ctx: IntegrationContext<Configuration>
+  client: Client
+  logger: IntegrationLogger
+  webhookUrl: string
+}) => {
+  const webhooks = await Promise.all(
+    ARR_OF_EVENTS.map(async (event) => {
+      const topic = getValue(event)
+      const webhookId = await createWebhook(topic, ctx, logger, webhookUrl)
+      return { webhookId, topic }
+    })
+  )
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏ Do you need topic? you dont use it anywhere. Since you only use the webhookIds and loop over them when you get state your state could be `string[]` instead of an object. Also you need to update your definition in `definitions/states.ts`

Reply:
I would like to keep it for reference.

Reply:
`Also you need to update your definition in definitions/states.ts`. Is that after I change it, or in the current code?

Reply:
Both, the definition is currently not in sync with the code. Also it's useless to keep unused code

Reply:
Sorry I don't understand. schema is expecting object, that's why I added object in the first place:
![image](https://github.com/botpress/botpress/assets/89806971/2e719a20-d47b-41a8-a97e-a0ec87ffa93d)